src/frontend/src/pages/AdminToolsPage/TeamsTools.tsx

```diff
@@ -0,0 +1,72 @@
+import { Box, Paper, Table, TableBody, TableCell, TableContainer, TableHead, TableRow } from '@mui/material';
+import PageBlock from '../../layouts/PageBlock';
+import { NERButton } from '../../components/NERButton';
+import { useAllTeams } from '../../hooks/teams.hooks';
+import LoadingIndicator from '../../components/LoadingIndicator';
+import ErrorPage from '../ErrorPage';
+import { fullNamePipe } from '../../utils/pipes';
+import CreateTeamModal from './CreateTeamModal';
+import { useState } from 'react';
+
+const TeamsTools = () => {
+  const [showCreateModal, setShowCreateModal] = useState(false);
+  const { data: allTeams, isLoading, isError, error } = useAllTeams();
+
+  if (!allTeams || isLoading) return <LoadingIndicator />;
+
+  if (isError) {
+    return <ErrorPage message={error.message} />;
+  }
+
+  const teamTableRows = allTeams.map((team) => (
```

Comment:
should abstract this table to take in an object thats like column names or something